aws/src/lint.js

```diff
@@ -51,18 +52,59 @@ class LIntHttpEnv {
   }
 }
 
-const sendHttpRequest = (options, postData) => new Promise((resolve, reject) => {
-  const req = https.request(options, (res) => {
-    const body = [];
-    res.on('data', chunk => body.push(chunk));
-    res.on('end', () => {
-      if (res.statusCode < 200 || res.statusCode >= 300) {
-        reject(new Error(`statusCode: ${res.statusCode}, response: ${Buffer.concat(body).toString()}`));
-      } else {
-        resolve('Log sent to Log Intelligence!');
-      }
-    });
+class LIntWFProxyEnv {
```

Comment:
Is creating explicit LIntWFProxyEnv related class necessary? Lambda should be agnostic of presence of Proxy. All it needs to know is whether to call http vs https end point, pass query params or not etc.
Can't we make changes without this class creation?

Reply:
Agreed. Refactor the code.